### PR TITLE
HubStatusServlet - use GET with query string instead of GET with body

### PR DESCRIPTION
**Pull request by tommywo**
- [X] By placing an `X` in the preceding checkbox, I verify that I have signed the [Contributor License Agreement](https://github.com/SeleniumHQ/selenium/blob/master/CONTRIBUTING.md#step-6-sign-the-cla)

### CODE REVIEW

**Comment by lukeis** [comment]
can you make it allow both POST data and url parameter? I'm a bit concerned here about breaking people unnecessarily.

@mach6

**Comment by mach6** [comment]
I share the concerns of @lukeis -- It should continue support (although it is odd from an HTTP protocol point of view) `GET with body` too.

One additional suggestion I have -- Can you make the default behavior when the query string is empty be return an _application/json_ document with all params known by the servlet (as the javadoc for the servlet states, but currently does not do)?  
++ _See javadoc line_ 58  -- `if no param is specified, all params known to the hub are returned.`

All said, personally I'm in the opinion that the `HubStatusServlet` needs to be deprecated in `3.0.0`  and replaced with a REST api for getting "hub" status which includes the information this servlet provides and more.

**Comment by tommywo** [comment]
It's now backward compatible with `GET with body`. 
I like the idea of REST API (i see there is pr open for it already -#265)

File: java/server/src/org/openqa/grid/web/servlet/HubStatusServlet.java [hunk 1]
```diff
@@ -101,7 +102,10 @@ private JsonObject getResponse(HttpServletRequest request) throws IOException {
       if (request.getInputStream() != null) {
         JsonObject requestJSON = getRequestJSON(request);
         List<String> keysToReturn = null;
-        if (requestJSON != null && requestJSON.has("configuration")) {
+
+        if (request.getParameter("configuration")!=null && !"".equals(request.getParameter("configuration"))) {
```
Comment (mach6) [inline java/server/src/org/openqa/grid/web/servlet/HubStatusServlet.java]: inconsistent use of whitespace here

File: java/server/test/org/openqa/grid/internal/StatusServletTests.java [hunk 2]
```diff
@@ -273,6 +274,34 @@ public void testHubGetSpecifiedConfig() throws IOException {
...
+  public void testHubGetSpecifiedConfigWithQueryString() throws IOException {
+
+    HttpClient client = httpClientFactory.getHttpClient();
+
+    ArrayList<String> keys = new ArrayList<>();
+    keys.add(URLEncoder.encode("timeout", "UTF-8"));
+    keys.add(URLEncoder.encode("I'm not a valid key", "UTF-8"));
+    keys.add(URLEncoder.encode("servlets", "UTF-8"));
+
+    String query = "?configuration="+String.join(",",keys);
```
Comment (mach6) [inline java/server/test/org/openqa/grid/internal/StatusServletTests.java]: inconsistent use of whitespace here

**Comment by tommywo** [comment]
@mach6 done

**Comment by tommywo** [comment]
@lukeis is it ok now? can you merge?

**Pull request merged**